kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java

```diff
@@ -273,4 +276,32 @@ private boolean isNetworkConfigurationPid(String pid) {
         }
         return result;
     }
+
+    private void updateNetInterfaces(Map<String, Object> configurationProperties, String pid) throws KuraException {
+        if (pid.equals(NETWORK_CONFIGURATION_SERVICE_PID)) {
+            Set<String> interfaceNames = new HashSet<>();
+            configurationProperties.keySet().forEach(key -> {
+                Optional<String> interfaceName = parseInterfaceName(key);
+                interfaceName.ifPresent(interfaceNames::add);
+            });
+            Map<String, Object> properties = this.configurationService.getComponentConfiguration(pid)
+                    .getConfigurationProperties();
+            String netInterfaces = (String) properties.get("net.interfaces");
+            List<String> netInterfacesList = Arrays.asList(netInterfaces.split(","));
+            for (String name : interfaceNames) {
+                if (!netInterfacesList.contains(name)) {
+                    netInterfaces += "," + name;
+                }
+            }
+            configurationProperties.put("net.interfaces", netInterfaces);
+        }
```

Comment:
You know I feel about early returns 😆 

```suggestion
        if (!pid.equals(NETWORK_CONFIGURATION_SERVICE_PID)) {
            return;
        }
        Set<String> interfaceNames = new HashSet<>();
        configurationProperties.keySet().forEach(key -> {
            Optional<String> interfaceName = parseInterfaceName(key);
            interfaceName.ifPresent(interfaceNames::add);
        });
        Map<String, Object> properties = this.configurationService.getComponentConfiguration(pid)
                .getConfigurationProperties();
        String netInterfaces = (String) properties.get("net.interfaces");
        List<String> netInterfacesList = Arrays.asList(netInterfaces.split(","));
        for (String name : interfaceNames) {
            if (!netInterfacesList.contains(name)) {
                netInterfaces += "," + name;
            }
        }
        configurationProperties.put("net.interfaces", netInterfaces);
```

... as always this suggestion might be subjective and therefore is not a blocker for merging.

Reply:
Or... you could bring that out to the outer caller:

```java
if (pid.equals(NETWORK_CONFIGURATION_SERVICE_PID)) {
    updateNetInterfaces(configurationProperties);
}
```

but that `for` is showing already too many indentation levels 🤔